code/game/objects/items/devices/painter/painter.dm

```diff
@@ -0,0 +1,70 @@
+/// Base painter object (Should not be spawned)
+/obj/item/painter
+	name = "modular painter"
+	icon = 'icons/obj/painting.dmi'
+	usesound = 'sound/effects/spray2.ogg'
+	flags = CONDUCT | NOBLUDGEON
+	w_class = WEIGHT_CLASS_SMALL
+	slot_flags = SLOT_BELT
+	materials = list(MAT_METAL = 5000, MAT_GLASS = 2000)
+
+	var/static/list/painter_types = list(
+		"Floor Painter" = /obj/item/painter/floor,
+		"Pipe Painter" = /obj/item/painter/pipe,
+		"Window Painter" = /obj/item/painter/pipe/window,
+		"Airlock Painter" = /obj/item/painter/airlock)
+
+/obj/item/painter/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Ctrl+click it in your hand to change the type!</span>"
+
+/obj/item/painter/CtrlClick(mob/user)
+	. = ..()
+	if(!ishuman(user)) // Only want human players doing this
+		return
+	if(loc != user) // Not being held
+		return
+
+	playsound(src, 'sound/effects/pop.ogg', 50, TRUE)
+	var/choice = show_radial_menu(user, src, painter_types, require_near = TRUE)
+	if(!choice)
+		return
+	var/choice_path = painter_types[choice]
+	if(istype(src, choice_path))
+		return
```

Comment:
There is an issue when switching from window painter to pipe painter. Since the pipe painter is a window painter subtype, this if is true, and will return.

Reply:
Ah damn, I thought I'd fixed that. I don't suppose there's some way of specifying that an object has to be one type specifically, and not a subtype?

Reply:
The easiest way would be to check the type directly: `if(type == choice_path)`

Reply:
Oh, that's easy.